File: ds lab/palindrome.c
/*This file implements a code to check if the given string is a palindrome using stack. */
#include<stdio.h>
#include<string.h>
int s[20],top=-1;
void push(char item)
{
        top++;
        s[top]=item;
}
void check(char str[20])
{
	int i,n=strlen(str);
	for(i=0;i<n/2;i++)
	{
		push(str[n-i-1]);
		if (str[i]!=s[i])
			break;
	}
	if (i==n/2)
		printf("palindrome\n");
	else
		printf("not palindrome\n");
}
int main()
{	char str[20];
	printf("enter string:");
	scanf("%s",str);
	check(str);
}

File: java_lab/Palindrome.java
public class Palindrome
{
   public static void main(String args[])
   {
     int j,i;
     char str[]={'m','a','l','a','y','a','l','a','m','\0'};
     for (i=0,j=8;i<j;i++,j--)
     {
	if (str[i]!=str[j])
	   break;
     }
     if (i==j)
	System.out.println("pallindrome");
     else
	System.out.println("not pallindrome");
   }
}
